django/db/models/lookups.py

```diff
@@ -554,3 +556,52 @@ class YearLte(YearComparisonLookup):
 
     def get_bound(self, start, finish):
         return finish
+
+
+class UUIDMixin:
+    '''Strip hyphens from a value when filtering a UUIDField'''
+
+    def process_lhs(self, qn, connection):
+        lhs, params = super().process_lhs(qn, connection)
+        lhs = "replace({}, '-', '')".format(lhs)
```

Comment:
I think we can run this only if database has native UUID field i.e. `connection.features.has_native_uuid_field` :thinking:

Reply:
Makes sense.

(To nitpick, is this enough? What if another backend comes along that has a native UUID field and stores values uppercase, hyphenated and/or surrounded by braces? These are other possible representations that the current code doesn't handle. So should we do this only explicitly for PostgreSQL?)